### ISSUE TITLE
How can I make villagers ask for emerald blocks when prices are over 64 emeralds? 

### ISSUE THREAD

**Issue by Gabo6480** [jacky8399/BalancedVillagerTrades#2]
For example, in theory an enchanted book of Efficiency V can cost up to 71 emeralds, however the price is capped at 64 emeralds.
I would like to allow villagers to ask for a roughly equivalent payment in emerald blocks instead of capping the price in emeralds when the prices exceed 64 emeralds.
 For the example of the Efficiency V enchanted book, ideally the villager would ask for 8 emerald blocks as payment for the enchanted book ( 71 / 9 = 7.888 rounding up cuz villagers are greedy ).

**Comment by jacky8399** [comment]
I don't think it is possible to recover the original price, since it is generated when the villager unlocked that trade. I can try to make the prices (min and max price) available and we'll see what we can do with that information.

**Comment by Gabo6480** [comment]
Hmmm I'm not sure, since I would also like to take into account the "Demand" and "Special Price" mechanics in the price change

**Comment by Gabo6480** [comment]
Well, maybe we can use the min and max price to "re-roll" the price of the item and then use it to replace the original one in the ItemStack of the MerchantRecipe, since it allows to store values higher than 64. Then we can have access to that value instead.

**Comment by jacky8399** [comment]
Thank you for the suggestion! This feature has been added in beta6 with this [example](https://github.com/jacky8399/BalancedVillagerTrades/wiki/Examples#make-librarians-optionally-charge-emerald-blocks-instead-of-capping-the-price-at-64-emeralds-since-20-beta6).
It rerolls level 5 enchantment books once, and if it costs more than 64 emeralds, the cost is replaced with emerald blocks instead.
From my own tests:
![bvt](https://user-images.githubusercontent.com/7542810/177588294-8620006e-3994-4982-86a5-0c63ab9a7ede.png)
(the item is renamed to avoid rerolling the cost every time a player interacts with the villager)

If you need help changing the cost formula to further "balance" the trades, feel free to leave a comment.

**Comment by Gabo6480** [comment]
I'm getting the following error:

[21:13:09 WARN]: [BalancedVillagerTrades] [Script Error] com.jacky8399.balancedvillagertrades.luaj.vm2.LuaError: result_item = trade.result
if result_item.name ~= null and result_item.name:find("Libro Encantado") then
        return
end

trade.result.name = "Libro Encantado Equisde":2 vm error: java.lang.NullPointerException: Cannot invoke "Object.toString()" because "value" is null

**Comment by jacky8399** [comment]
Sorry, this has been fixed in beta 7
